### DISCUSSION TITLE
How to differentiate between connected handles and un-connected ones?

### DISCUSSION THREAD

**Discussion by samuellawerentz** [xyflow/xyflow#2194] [Q&A]
Hi,
I was just wondering if we could have two different colors - one for handles that are connected to other nodes and the other color for handles that are not connected to any node.

One solution is to update the color of the handle during the `onConnect` event but the catch here is, I'm rendering a readonly flow diagram where user can only view and cannot interact. 

This flow diagram has some nodes that are connected and some that are not connected, now I need to differentiate these two during the initial render.

Any help would be of great use!! 
Thanks

**Comment by moklick** [comment]
Hey @samuellawerentz 

I would create a custom node that passes a class name or style to its `<Handle />` components when they are connected.
